### PR TITLE
Replace TravisCI with GitHub Workflows

### PR DESCRIPTION
**Pull request by HaSistrunk**
Replaces TravisCI with GitHub Workflow tests and deploys.

Includes separate deploy files for the Style Guide (which is the Storybook site that serves to styles.rockarch.org) and the Style Library (which is our compiled CSS stylesheet and associated fonts and image assets, versioned to match the tagged release in GitHub). The Style Guide should deploy on pushes to base, and the Style Library should only deploy when a new release is tagged on the base branch. 

Also updates node version in depedency updates workflow.

### CODE REVIEW

File: .github/workflows/dependencies.yml
Comment (p-galligan) [inline .github/workflows/dependencies.yml:1]: This needs to be updated to use the GitHub token generator app.

**Review by p-galligan — CHANGES_REQUESTED** [review]

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -14,21 +13,30 @@ jobs:
...
         with:
-          ref: base
+          token: ${{ steps.app-token.outputs.token }}
+          ref: development
+          persist-credentials: false
 
       - uses: actions/setup-node@v3
         with:
-          node-version: '18'
+          node-version: lts/iron
```
Comment (helrond) [inline .github/workflows/dependencies.yml]: Can we not use the same version string we're using in workflows below: 'lts/*'?

Comment (HaSistrunk) [reply .github/workflows/dependencies.yml]: Yeah, that makes sense. I copied that from DIMES, which... maybe also needs an update?

Comment (p-galligan) [reply .github/workflows/dependencies.yml]: Yeah, I don't have strong feelings, but would argue that `lts/*` makes more sense to me.

**Review by ioduok — APPROVED** [review]

File: .github/workflows/dependencies.yml (hunk 1 above)
Comment (helrond) [reply .github/workflows/dependencies.yml]: Consistency is probably good. @p-galligan can you take a look at DIMES?

Comment (p-galligan) [reply .github/workflows/dependencies.yml]: > Consistency is probably good. @p-galligan can you take a look at DIMES?

This is done!

**Review by helrond — APPROVED** [review]

**Review by p-galligan — APPROVED** [review]

**Pull request merged**